src/charm.py

```diff
@@ -92,6 +92,20 @@ def __init__(self, *args):
         super().__init__(*args)
         self._stored.set_default(config_hash=None, launched_with_peers=False)
 
+        if url := self.model.config.get("web_external_url"):
+            extra_sans_dns = [cast(str, urlparse(url).netloc)]
+        else:
+            extra_sans_dns = None
```

Comment:
Just a matter of taste ;-) 

```suggestion
        extra_sans_dns = None
        if url := self.model.config.get("web_external_url"):
            extra_sans_dns = [cast(str, urlparse(url).netloc)]
```